.github/workflows/build_env_run_tests.yml

```diff
@@ -47,22 +47,44 @@ jobs:
       fail-fast: false
     defaults:
       run:
-        shell: bash -l {0}  # reset shell for each step
+        shell: ${{ matrix.os == 'windows-latest' && 'cmd' || 'bash' }} -l {0}
     steps:
-      - name: checkout repo
+      - name: Checkout repo
         uses: actions/checkout@v2
-      - name: set up conda env
+
+      - name: Set up conda env (Linux, Windows)
+        if: ${{ matrix.os != 'macos-latest' }}
         uses: conda-incubator/setup-miniconda@v2
         with:
           use-mamba: true
           miniforge-variant: Mambaforge
           python-version: ${{ matrix.python-version }}
           environment-file: ./env_config/env.yml
           activate-environment: aeon
+
+      - name: Set up conda env (macOS)
+        if: ${{ matrix.os == 'macos-latest' }}
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          use-mamba: true
+          miniforge-variant: Mambaforge
+          python-version: ${{ matrix.python-version }}
+          environment-file: ./env_config/env_macos.yml
+          activate-environment: aeon
+          architecture: arm64
+          miniconda-version: "latest"
+
+      - name: Install datajoint wheel build with pip flag (macOS)
+        if: ${{ matrix.os == 'macos-latest' }}
+        run: |
+          source $CONDA/bin/activate aeon
+          pip install --use-pep517 datajoint git+https://github.com/datajoint-company/datajoint-utilities.git
```

Comment:
I guess this partly answers my earlier comment. Would it work if we just use `--use-pep517` to setup the environment for all OSes?

Reply:
To be resolved in #402